File: client-clojure/src/sirqul_io_t_platform/specs/listing.clj
(ns sirqul-io-t-platform.specs.listing
  (:require [clojure.spec.alpha :as s]
            [spec-tools.data-spec :as ds]
            [sirqul-io-t-platform.specs.account :refer :all]
            [sirqul-io-t-platform.specs.participant :refer :all]
            [sirqul-io-t-platform.specs.filter :refer :all]
            )
  (:import (java.io File)))


(def listing-data
  {
   (ds/opt :id) int?
   (ds/opt :active) boolean?
   (ds/opt :valid) boolean?
   (ds/opt :owner) account-spec
   (ds/opt :name) string?
   (ds/opt :description) string?
   (ds/opt :start) inst?
   (ds/opt :end) inst?
   (ds/opt :locationName) string?
   (ds/opt :locationDescription) string?
   (ds/opt :participants) (s/coll-of participant-spec)
   (ds/opt :externalId) string?
   (ds/opt :externalGroupId) string?
   (ds/opt :filters) (s/coll-of filter-spec)
   (ds/opt :private) boolean?
   })

(def listing-spec
  (ds/spec
    {:name ::listing
     :spec listing-data}))

File: client-scalaz/src/main/scala/org/openapitools/client/api/Listing.scala
package org.openapitools.client.api

import argonaut._
import argonaut.EncodeJson._
import argonaut.DecodeJson._

import org.http4s.{EntityDecoder, EntityEncoder}
import org.http4s.argonaut._
import org.joda.time.DateTime

import java.time.OffsetDateTime

import Listing._

case class Listing (
  id: Option[Long],
active: Option[Boolean],
valid: Option[Boolean],
owner: Option[Account],
name: Option[String],
description: Option[String],
start: Option[OffsetDateTime],
end: Option[OffsetDateTime],
locationName: Option[String],
locationDescription: Option[String],
participants: Option[List[Participant]],
externalId: Option[String],
externalGroupId: Option[String],
filters: Option[List[Filter]],
`private`: Option[Boolean])

object Listing {
  import DateTimeCodecs._

  implicit val ListingCodecJson: CodecJson[Listing] = CodecJson.derive[Listing]
  implicit val ListingDecoder: EntityDecoder[Listing] = jsonOf[Listing]
  implicit val ListingEncoder: EntityEncoder[Listing] = jsonEncoderOf[Listing]
}
